crates/sui/src/client_commands.rs

```diff
@@ -1459,31 +1729,50 @@ impl SuiClientCommands {
                 count,
                 gas,
                 gas_budget,
+                dry_run,
                 serialize_unsigned_transaction,
                 serialize_signed_transaction,
             } => {
                 let signer = context.get_object_owner(&coin_id).await?;
                 let client = context.get_client().await?;
-                let data = match (amounts, count) {
-                    (Some(amounts), None) => {
-                        client
-                            .transaction_builder()
-                            .split_coin(signer, coin_id, amounts, gas, gas_budget)
-                            .await?
-                    }
-                    (None, Some(count)) => {
-                        if count == 0 {
-                            return Err(anyhow!("Coin split count must be greater than 0"));
-                        }
-                        client
-                            .transaction_builder()
-                            .split_coin_equal(signer, coin_id, count, gas, gas_budget)
-                            .await?
-                    }
-                    _ => {
-                        return Err(anyhow!("Exactly one of `count` and `amounts` must be present for split-coin command."));
+                if amounts.is_some() && count.is_some() {
+                    return Err(anyhow!("Exactly one of `count` and `amounts` must be present for split-coin command."));
+                }
+                if let Some(count) = count {
+                    if count == 0 {
+                        return Err(anyhow!("Coin split count must be greater than 0"));
                     }
-                };
+                }
```

Comment:
nit (optional): Could be a single `match` command (also what happens if they are both `None`?

```rust
match (count, amounts) {
  (Some(0), _) => bail!("Coin split count must be greater than 0"),
  (Some(_), Some(_)) | (None, None) => bail!("Exactly one of `count` and `amounts`...");
  (Some(_), None) | (None, Some(_)) => { /* nop */ },
``

Reply:
Oops, forgot that case.